tpchgen-cli/src/plan.rs

```diff
@@ -60,59 +60,98 @@ impl GenerationPlan {
     /// Returns the number of parts to generate
     ///
     /// cli_part and cli_part_count are passed from the CLI arguments
-    /// to specify a particular part or number of parts to generate.
-    pub fn new(
+    /// to specify a particular part or number of partitions to generate.
+    pub fn try_new(
         table: &Table,
         format: OutputFormat,
         scale_factor: f64,
-        cli_part: i32,
-        cli_part_count: i32,
+        cli_part: Option<i32>,
+        cli_part_count: Option<i32>,
         num_threads: usize,
-    ) -> Self {
+    ) -> Result<Self, String> {
         // If a single part is specified, split it into chunks to enable parallel generation.
-        if cli_part != -1 || cli_part_count != -1 {
-            // These tables are small not parameterized by part count,
-            // so we must create only a single part.
-            if table == &Table::Nation || table == &Table::Region {
-                return Self {
-                    part_count: 1,
-                    part_list: vec![1],
-                };
+        match (cli_part, cli_part_count) {
+            (Some(_part), None) => Err(String::from(
+                "The --part option requires the --parts option to be set",
+            )),
+            (None, Some(_part_count)) => {
+                // TODO automatically create multiple files if part_count > 1
+                // and part is not specified
```

Comment:
I also filed a ticket to track this
- https://github.com/clflushopt/tpchgen-rs/issues/167

```suggestion
                // and part is not specified
                // https://github.com/clflushopt/tpchgen-rs/issues/167
```